pennylane_lightning/lightning_tensor/lightning_tensor.py

```diff
@@ -0,0 +1,343 @@
+        c_dtype: Datatypes for the tensor representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        **kwargs: keyword arguments. The following options are currently supported:
+
+            ``max_bond_dim`` (int): Maximum bond dimension for the MPS simulator.
+                It corresponds to the number of Schmidt coefficients retained at the end of the SVD algorithm when applying gates. Default is ``None``.
+            ``cutoff`` (float): Truncation threshold for the Schmidt coefficients in a MPS simulator. Default is ``1e-16``.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    # So far we just consider the options for MPS simulator
+    _device_options = (
+        "backend",
+        "c_dtype",
+        "cutoff",
+        "method",
+        "max_bond_dim",
+    )
+
+    _new_API = True
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        *,
+        wires=None,
+        backend="quimb",
+        method="mps",
+        shots=None,
+        c_dtype=np.complex128,
+        **kwargs,
+    ):
+
+        if not accepted_backends(backend):
+            raise ValueError(f"Unsupported backend: {backend}")
+
+        if not accepted_methods(method):
+            raise ValueError(f"Unsupported method: {method}")
+
+        if shots is not None:
+            raise ValueError("LightningTensor does not support finite shots.")
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._num_wires = len(self.wires) if self.wires else 0
+        self._backend = backend
+        self._method = method
+        self._c_dtype = c_dtype
+
+        # options for MPS
+        self._max_bond_dim = kwargs.get("max_bond_dim", None)
+        self._cutoff = kwargs.get("cutoff", 1e-16)
+
+        self._interface = None
+        interface_opts = self._setup_execution_config().device_options
+
+        if self.backend == "quimb" and self.method == "mps":
+            self._interface = QuimbMPS(
+                self._num_wires,
+                interface_opts,
+                self._c_dtype,
+            )
+
+        for arg in kwargs:
+            if arg not in self._device_options:
+                raise TypeError(
+                    f"Unexpected argument: {arg} during initialization of the LightningTensor device."
+                )
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "lightning.tensor"
+
+    @property
+    def num_wires(self):
+        """Number of wires addressed on this device."""
+        return self._num_wires
+
+    @property
+    def backend(self):
+        """Supported backend."""
+        return self._backend
+
+    @property
+    def method(self):
+        """Supported method."""
+        return self._method
+
+    @property
+    def c_dtype(self):
+        """Tensor complex data type."""
+        return self._c_dtype
+
+    dtype = c_dtype
```

Comment:
Is this meant to be class or instance defined?

Reply:
To be honest, I wasn't sure since we still don't have C++ backends. I proceeded as currently done for `lightning.qubit`, but we can adjust this in the future after the first `CutensorNet` prototype is ready to be merged

Reply:
> Is this meant to be class or instance defined?

Instance defined. `c_dtype` is a property.